python/tk_framework_desktopserver/shotgun/__init__.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+# We want class-level variables to persist, so we'll setup a cache here
+# that the factory functuon below populates.
```

Comment:
Because the factory function imports the api modules in its local scope, subsequent calls to the factory won't return instances of the same class definition unless those are promoted to the global scope somehow. I'm achieving that with a module-level cache.